core/blockchain.go

```diff
@@ -2498,3 +2505,19 @@ func (bc *BlockChain) SetBlockValidatorAndProcessorForTesting(v Validator, p Pro
 func (bc *BlockChain) SetTrieFlushInterval(interval time.Duration) {
 	bc.flushInterval.Store(int64(interval))
 }
+
+// RealizedSecondsPerBlock computes the average seconds per block for the last n blocks
+// if n <= 0 the BlocksPerSecondCheckLength is used
+func (bc *BlockChain) RealizedSecondsPerBlock(n int64) (float64, error) {
```

Comment:
One thing we do, in certain places, is use a moving average which works like this: 
```
 value = value * 0.9 + new_value*0.1
```
The constants `0.9` and `0.1` can be tuned, but it's a pretty neat implementation of a moving average. You woudn't care about old values, in the beginning it would be not trustworthy, but after a while it would catch up. 

Might be worth using here, for simplicity? Also, we prefer lower-case (unexported) methods unless there's some explicit need for making the methods exported.